weeklyChallenges04/ChallengesSet04.cs

```diff
@@ -1,52 +1,127 @@
 ﻿using System;
+using System.Collections.Generic;
 namespace weeklyChallenges04
 {
     public class ChallengesSet04
     {
         public int AddEvenSubtractOdd(int[] numbers)
         {
-            throw new NotImplementedException();
+            int z = 0;
+            foreach (int x in numbers)
+            {
+                if (x % 2 == 0)
+                    z += x;
+                else
+                    z -= x;
+            }
+            return z;
         }
 
         public int GetLengthOfShortestString(string str1, string str2, string str3, string str4)
         {
-            throw new NotImplementedException();
+            int strl1 = str1.Length;
+            int strl2 = str2.Length;
+            int strl3 = str3.Length;
+            int strl4 = str4.Length;
+            int shortest = 0;
+            if (strl1 > strl2)
+                shortest = strl2;
+            else
+                shortest = strl1;
+            if (shortest > strl3)
+                shortest = strl3;
+            if (shortest > strl4)
+                shortest = strl4;
+            return shortest;
         }
 
         public int GetSmallestNumber(int number1, int number2, int number3, int number4)
         {
-            throw new NotImplementedException();
+            int smallest = 0;
+            if (number1 > number2)
+                smallest = number2;
+            else
+                smallest = number1;
+            if (smallest > number3)
+                smallest = number3;
+            if (smallest > number4)
+                smallest = number4;
+            return smallest;
         }
 
         public void ChangeBusinessNameTo_TrueCoders(Business biz)
         {
-            throw new NotImplementedException();
+            biz.Name = "TrueCoders";
         }
 
         public bool CouldFormTriangle(int sideLength1, int sideLength2, int sideLength3)
         {
-            throw new NotImplementedException();
+            if (sideLength1 + sideLength2 > sideLength3 && sideLength1 + sideLength3 > sideLength2 && sideLength2 + sideLength3 > sideLength1)
+                return true;
+            else
+                return false;
```

Comment:
Since this is a bool, it could've been simply returned because the result will either be true/false:

`return (sideLength1 + sideLength2 > sideLength3 && sideLength1 + sideLength3 > sideLength2 && sideLength2 + sideLength3 > sideLength1);`



return